### ISSUE TITLE
Excluded is not working properly then files are passed directly

### ISSUE THREAD

**Issue by Vadim-Lavrov** [realm/SwiftLint#6044]
### New Issue Checklist

- [X] I've Updated SwiftLint to the latest version.
- [X] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

Im using swiftlint as part of CI process 
Im collecting modified files from pull request and lint them:

```bash
     MODIFIED_FILES=$(echo "$ALL_CHANGES" | grep -E '\.swift' || true)

      if [ -n "$MODIFIED_FILES" ]; then
        echo "Running swiftlint on modified files..."
        swiftlint lint --config .swiftlint.yml --reporter checkstyle -- $MODIFIED_FILES > swiftlint-report.xml
      else
        echo "No Swift files were modified"
        touch swiftlint-report.xml && echo "[]" > swiftlint-report.xml
      fi
```

Some of those files are generated. 
 
They are NOT getting excluded. Patter matching (*) is not working for whose files, if I pass them as parameter. 
Only simple paths are working fine , like:
 - Packages/Sources/Generated/AutoMockable.generated.swift  
or 
 - Packages/RetenoSDK/Sources/Generated/
 
If files are not passed and linter is linting whole project - everything is working as expected

### Environment

* SwiftLint version 0.58.2
* Xcode version 16.2
* Installation method used Homebrew
* Configuration file:

```yml
   excluded:
  - "**/*.generated.swift"
  - "**/*.generated1.swift"
  - "**/*.generated2.swift"
  - "**/Generated/*.swift"
  - "**/Mocks/**"

identifier_name:
  min_length: 2
  max_length: 70
  validates_start_with_lowercase: true
  allowed_symbols:
    - "_"
  excluded:
    - id
    - URL
    - GlobalAPIKey
    - UI
    - to
    - vm
    - vc

analyzer_rules:
  - unused_import
```

No nested configs

**Comment by SimplyDanny** [comment]
Did you check if using the `--force-exclude` helps? You definitely need to add it to your command. The question is whether it actually works in your specific case.

**Comment by Vadim-Lavrov** [comment]
@SimplyDanny It really helped, thank you.
